### PR TITLE
Create instrumentation-core apache-httpclient-4.0

### PR DESCRIPTION
**Pull request by mabdinur**
Extract code that can be used in manual instrumentation from instrumentation:apache-httpclient:apache-httpclient-4.0 to instrumentation-core.

### CODE REVIEW

File: instrumentation-core/spring/spring-webflux-5.0/spring-webflux-5.0.gradle [hunk 1]
```diff
@@ -9,7 +9,7 @@ ext {
 
 apply from: "$rootDir/gradle/java.gradle"
 
-group = 'io.opentelemetry.instrumentation.spring.webflux'
+group = 'io.opentelemetry.instrumentation.spring'
```
Comment (trask) [inline instrumentation-core/spring/spring-webflux-5.0/spring-webflux-5.0.gradle:12]: I think let's not create maven subgroups, and put manual instrumentation directly in `io.opentelemetry.instrumentation` (which matches what we are doing in `io.opentelemetry.instrumentation.auto`)
```suggestion
group = 'io.opentelemetry.instrumentation'
```

Comment (mabdinur) [reply instrumentation-core/spring/spring-webflux-5.0/spring-webflux-5.0.gradle]: I'll make this change in the spring-web and spring-webmvc projects as well

**Comment by mabdinur** [comment]
Muzzle is failing on last commit.

Muzzle passed for all commits except the last one (https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/694/commits/7d02d9dbff4d28a34dcb8910fbb6b54dee5100e5). I did not change any gradle files in this commit so I'm not sure why `:instrumentation:apache-httpclient:apache-httpclient-4.0:muzzle-AssertPass` tests are failing. Has anyone seen this issue before?

**Comment by iNikem** [comment]
Make your `ApacheHttpClientDecorator` public. It is called directly from instrumented class, so should be reachable from library code.

**Comment by trask** [comment]
Hey @mabdinur! The `muzzle` issue is because of the static `TRACER` field in `ApacheHttpClientInstrumentation`.

The references to that `TRACER` field then pull in the whole of `ApacheHttpClientInstrumentation`, which then pulls in those bytebuddy classes. This is a common issue I need to doc in #536.

I suggest having overloaded methods in `ApacheHttpClientHelper` that either take a `Tracer` or use the default one, e.g.

```
public static SpanWithScope doMethodEnter(HttpUriRequest request, Tracer tracer)
```

and

```
public static SpanWithScope doMethodEnter(HttpUriRequest request)
```

This will be useful for people using manual instrumentation, and also can be used then from auto instrumentation and you can remove the `TRACER` field from the auto instrumentation.

File: instrumentation-core/apache-httpclient-4.0/src/main/java/io/opentelemetry/instrumentation/apachehttpclient/v4_0/ApacheHttpClientDecorator.java [hunk 2]
```diff
@@ -19,19 +19,15 @@
...
 import io.opentelemetry.auto.bootstrap.instrumentation.decorator.HttpClientDecorator;
-import io.opentelemetry.trace.Tracer;
 import java.net.URI;
 import org.apache.http.Header;
 import org.apache.http.HttpMessage;
 import org.apache.http.HttpResponse;
 import org.apache.http.client.methods.HttpUriRequest;
 
-class ApacheHttpClientDecorator extends HttpClientDecorator<HttpUriRequest, HttpResponse> {
+public class ApacheHttpClientDecorator extends HttpClientDecorator<HttpUriRequest, HttpResponse> {
```
Comment (trask) [inline instrumentation-core/apache-httpclient-4.0/src/main/java/io/opentelemetry/instrumentation/apachehttpclient/v4_0/ApacheHttpClientDecorator.java:28]: Can you try again with this package-private, now that `muzzle` is fixed? Better not to expose this to manual instrumenters if possible.
```suggestion
class ApacheHttpClientDecorator extends HttpClientDecorator<HttpUriRequest, HttpResponse> {
```

File: instrumentation-core/apache-httpclient-4.0/src/main/java/io/opentelemetry/instrumentation/apachehttpclient/v4_0/HttpHeadersInjectAdapter.java [hunk 3]
```diff
@@ -19,7 +19,7 @@
 import io.opentelemetry.context.propagation.HttpTextFormat;
 import org.apache.http.client.methods.HttpUriRequest;
 
-class HttpHeadersInjectAdapter implements HttpTextFormat.Setter<HttpUriRequest> {
+public class HttpHeadersInjectAdapter implements HttpTextFormat.Setter<HttpUriRequest> {
```
Comment (trask) [inline instrumentation-core/apache-httpclient-4.0/src/main/java/io/opentelemetry/instrumentation/apachehttpclient/v4_0/HttpHeadersInjectAdapter.java]: same here
```suggestion
class HttpHeadersInjectAdapter implements HttpTextFormat.Setter<HttpUriRequest> {
```

Comment (mabdinur) [reply instrumentation-core/apache-httpclient-4.0/src/main/java/io/opentelemetry/instrumentation/apachehttpclient/v4_0/HttpHeadersInjectAdapter.java]: Done

**Pull request merged**